### ISSUE TITLE
Issue with external audio device

### ISSUE THREAD

**Issue by marcdsdavid** [glicol/glicol-cli#21]
Not sure exactly what is the cause of this, but when I run the cli tool the audio is glitched (high pitched) and also comes out of all the outputs of my device.

https://github.com/glicol/glicol-cli/assets/7226788/2be15f6d-4973-46e3-82de-f94d65c55e63

**Comment by chaosprint** [comment]
Can you try to clone this repo, change the BLOCK_SIZE to 64 (Line 34 at `main.rs`) and run with `cargo run --release`?
Also try to change the content of `test.glicol` to `o: sin 440`. Hopefully you can get a pure sine tone.

**Comment by marcdsdavid** [comment]
Just tried your recommended changes but I still get a very high pitched tone that gets sent to all the software playback channels of my soundcard (RME Babyface Pro fs) - I also tried a sampling rate of 44.1kHz but it gives me the same result.

It works completely fine if I set my default output device to my Macbook Pro speakers.

here is what I did to get the result showed in the video:

- Cloned the repo, had to run cargo update because of this error (error: failed to run custom build command for `coreaudio-sys v0.2.11`)
- Changed the BLOCK_SIZE to 64 as you mentioned

and ran `cargo run --release test.glicol`


https://github.com/glicol/glicol-cli/assets/7226788/8fe8d9c3-9977-4645-84db-a5413e8560cf

**Comment by chaosprint** [comment]
> Just tried your recommended changes but I still get a very high pitched tone that gets sent to all the software playback channels of my soundcard (RME Babyface Pro fs) - I also tried a sampling rate of 44.1kHz but it gives me the same result.
> 
> It works completely fine if I set my default output device to my Macbook Pro speakers.
> 
> here is what I did to get the result showed in the video:
> 
> * Cloned the repo, had to run cargo update because of this error (error: failed to run custom build command for `coreaudio-sys v0.2.11`)
> * Changed the BLOCK_SIZE to 64 as you mentioned
> 
> and ran `cargo run --release test.glicol`
> 
>  CleanShot.2023-11-01.at.10.57.29.mp4

right. can you try to clone this repo: https://github.com/RustAudio/cpal/

and run `cargo run --release --example beep`

glicol cli depends on `cpal` to access the audio interface

**Comment by marcdsdavid** [comment]
cpal gives me a nice 440Hz tone. It sounds distorted on the recording probably because it still sends the audio to some extra software outputs when it should just go to AN 1/2 and I assume the screen recorder captures them all.


https://github.com/glicol/glicol-cli/assets/7226788/eb6bfa88-2d97-47c6-ac8f-3b276a895b08

**Comment by chaosprint** [comment]
> cpal gives me a nice 440Hz tone. It sounds distorted on the recording probably because it still sends the audio to some extra software outputs when it should just go to AN 1/2 and I assume the screen recorder captures them all.
> 
>  CleanShot.2023-11-01.at.12.26.07.mp4

thanks for the info. I will look into it with my Focusrite audio interface when I get home. You may try the first method one more time if possible but change the block size to 32.

**Comment by marcdsdavid** [comment]
Thanks! I tried with a block size of 32 and get the same result as using any other block size (instead of 440Hz I get 3080Hz tone)
